src/pages/lessons/Lessons.tsx

```diff
@@ -8,17 +8,14 @@ import CustomLessonSetup from "./custom/CustomLessonSetup";
 import Loadable from "react-loadable";
 import PageLoading from "../../components/PageLoading";
 
-type LessonsRoutingProps = {
-  customiseLesson: () => void;
-  customLesson: any;
-  generateCustomLesson: any;
-  handleLesson: any;
-  lesson: any;
-  lessonIndex: any;
-  match: any;
-  stopLesson: any;
-  [key: string]: any;
-};
+type Optional<T, K extends keyof T> = Pick<Partial<T>, K> & Omit<T, K>;
+type LessonsRoutingProps = Optional<
+  & RouteComponentProps
+  & ComponentProps<typeof Lesson>
+  & ComponentProps<typeof CustomLessonSetup>
+  & ComponentProps<typeof AsyncCustomLessonGenerator>
+  // TODO: check this. it's not passed from parent
+  , "lessonLength">
```

Comment:
For now I marked `// @ts-expect-error` for this. I can make them optional if that's better